product/src/main/java/com/yas/product/repository/ProductOptionValueRepository.java

```diff
@@ -3,11 +3,15 @@
 
 import com.yas.product.model.Product;
 import com.yas.product.model.ProductOptionValue;
+import com.yas.product.viewmodel.productoption.ProductOptionValueProjection;
 import java.util.List;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public interface ProductOptionValueRepository extends JpaRepository<ProductOptionValue, Long> {
     List<ProductOptionValue> findAllByProduct(Product product);
+    
+    List<ProductOptionValueProjection> findAllProjectedByProduct(Product product);
```

Comment:
1. Yes, we can, the new function is an optimization to reduce the number of select field of JPA SQL call, because `product` field is eagerly fetched, which I don't need.
2. I don't know how to answer this one. Do we have a rule about that? If so, I will revert back to the old function.